src/toil/statsAndLogging.py

```diff
@@ -112,9 +112,10 @@ def callback(fileHandle):
                 pass
             else:
                 def logWithFormatting(jobStoreID, jobLogs):
-                    logFormat = '\n%s    ' % jobStoreID
                     logger.debug('Received Toil worker log. Disable debug level '
-                                 'logging to hide this output\n%s', logFormat.join(jobLogs))
+                                 'logging to hide this output')
+                    for line in jobLogs:
+                        logger.debug('%s    %s', jobStoreID, line)
```

Comment:
There is also another place in the code where the full log is read into memory (leader.py line 730). I've been wanting to refactor the worker log related code for a while so I'm happy to merge this as is & I can fix the other location tomorrow when I consolidate the methods. 

Let me know if that works for you!